cos-gpu-installer-docker/entrypoint.sh

```diff
@@ -274,82 +284,50 @@ configure_nvidia_installation_dirs() {
-  echo "$1" | cut -d "." -f 2
-}
-
-installer_default_download_url() {
-  if (( $(major_version "${NVIDIA_DRIVER_VERSION}") < 390 )); then
-    # Versions prior to 390 are downloaded from the upstream location.
-    info "Downloading Nvidia installer from https://us.download.nvidia.com/... "
-    echo "https://us.download.nvidia.com/tesla/${NVIDIA_DRIVER_VERSION}/NVIDIA-Linux-x86_64-${NVIDIA_DRIVER_VERSION}.run"
-    return
-  fi
-
-  info "Downloading Nvidia installer from https://storage.googleapis.com/... "
-  # projects/000000000000/zones/us-west1-a -> us
-  local -r instance_location="$(curl -sfS "http://metadata.google.internal/computeMetadata/v1/instance/zone" -H "Metadata-Flavor: Google" | cut -d '/' -f4 | cut -d '-' -f1)"
-  declare -A location_mapping
-  location_mapping=( ["us"]="us" ["asia"]="asia" ["europe"]="eu" )
-  # Use us as default download location.
-  local -r download_location="${location_mapping[${instance_location}]:-us}"
-
-  if (( $(major_version "${NVIDIA_DRIVER_VERSION}") == 390 )); then
-    # The naming format changed after version 390.
-    echo "https://storage.googleapis.com/nvidia-drivers-${download_location}-public/TESLA/NVIDIA-Linux-x86_64-${NVIDIA_DRIVER_VERSION}.run"
-  elif (( $(major_version "${NVIDIA_DRIVER_VERSION}") >= 396 )) && (( $(minor_version "${NVIDIA_DRIVER_VERSION}") >= 37 )); then
-    # Apparently the naming format changed again starting since 396.37.
-    echo "https://storage.googleapis.com/nvidia-drivers-${download_location}-public/tesla/${NVIDIA_DRIVER_VERSION}/NVIDIA-Linux-x86_64-${NVIDIA_DRIVER_VERSION}.run"
-  else
-    echo "https://storage.googleapis.com/nvidia-drivers-${download_location}-public/tesla/${NVIDIA_DRIVER_VERSION}/NVIDIA-Linux-x86_64-${NVIDIA_DRIVER_VERSION}-diagnostic.run"
-  fi
-
-}
-
-get_nvidia_installer_url() {
-  if [ ! -v NVIDIA_DRIVER_DOWNLOAD_URL ]; then
-    NVIDIA_DRIVER_DOWNLOAD_URL="$(installer_default_download_url)"
-  fi
-  echo "${NVIDIA_DRIVER_DOWNLOAD_URL}"
-}
-
-get_nvidia_installer_runfile() {
-  # Cache NVIDIA_INSTALLER_RUNFILE value to avoid multiple metadata server
-  # access.
-  if [ ! -v NVIDIA_INSTALLER_RUNFILE ]; then
-    local -r nvidia_driver_download_url="$(get_nvidia_installer_url)"
-    NVIDIA_INSTALLER_RUNFILE="$(basename ${nvidia_driver_download_url})"
-  fi
-  echo ${NVIDIA_INSTALLER_RUNFILE}
-}
-
 download_nvidia_installer() {
-  info "Downloading Nvidia installer ... "
+  info "Downloading GPU installer ... "
+  . gpu_installer_url_lib.sh
   pushd "${NVIDIA_INSTALL_DIR_CONTAINER}"
-  local -r nvidia_driver_download_url="$(get_nvidia_installer_url)"
-  info "Downloading from ${nvidia_driver_download_url}"
-  curl -L -sS "${nvidia_driver_download_url}" -o "$(get_nvidia_installer_runfile)"
+  local -r gpu_installer_download_url="$(get_gpu_installer_url ${NVIDIA_DRIVER_VERSION} ${VERSION_ID} ${BUILD_ID})"
+  info "Downloading from ${gpu_installer_download_url}"
+  INSTALLER_FILE="$(basename ${gpu_installer_download_url})"
```

Comment:
Would it make sense to initialize INSTALLER_FILE at the top to indicate that it's global?

Reply:
Done